src/test/java/io/personium/core/bar/BarFileExporterTest.java

```diff
@@ -0,0 +1,322 @@
+        // box/file1.json
+        MockDavCmpFsImpl col = new MockDavCmpFsImpl("col", dcBox, acl1);
+        new MockDavCmpFsImpl("file1.json", col, acl1, DavCmp.TYPE_DAV_FILE);
+        new MockDavCmpFsImpl("file2.json", dcBox, null, DavCmp.TYPE_DAV_FILE);
+
+        AccessContext ac = null;
+        CellRsCmp cellCmp = new CellRsCmp(dcCell, box.getCell(), ac);
+        return new BoxRsCmp(cellCmp, dcBox, ac, box);
+    }
+
+    public static byte[] responseToBytes(StreamingOutput so) {
+        byte [] byteArray = null;
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()){
+            so.write(baos);
+            byteArray = baos.toByteArray();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return byteArray;
+    }
+    public static void readExportedBarFile(byte[] barBytes) {
+        barZipEntryList = new ArrayList<>();
+        barZipEntryMap = new HashMap<>();
+        barZipContentMap = new HashMap<>();
+        try (
+            ByteArrayInputStream bais = new ByteArrayInputStream(barBytes);
+            ZipInputStream zis = new ZipInputStream(bais)
+        ){
+            ZipEntry ent = null;
+            int bufferSize = 1024;
+            while((ent = zis.getNextEntry()) != null) {
+                String entName = ent.getName();
+                log.info(entName);
+
+                barZipEntryList.add(ent);
+                barZipEntryMap.put(entName, ent);
+
+                byte data[] = new byte[bufferSize];
+                int count = 0;
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                while((count = zis.read(data, 0, bufferSize)) != -1) {
+                    baos.write(data,0,count);
+                }
+
+                barZipEntryMap.put(entName, ent);
+                barZipContentMap.put(entName, baos.toByteArray());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void prepareBarFile() {
+        // prepare mocks for testing
+        Cell cell = mockCell(CELL_URL);
+        Box box = mockBox(cell, BOX_NAME, BOX_SCHEMA_URL);
+        BoxRsCmp boxRsCmpMock = mockBoxRsComp(box);
+
+        // create bar file response
+        BarFileExporter exporter =  new BarFileExporter(boxRsCmpMock);
+        //  ... this part is the core of this unit test ...
+        Response res = exporter.export();
+
+        // parse the response bar file contents
+        StreamingOutput so = (StreamingOutput)res.getEntity();
+        byte[] barFileBytes = responseToBytes(so);
+        readExportedBarFile(barFileBytes);
+    }
+
+    @Before
+    public void before() throws Exception {
+    }
+
+    /**
+     * bar file composition check.
+     */
+    @Test
+    public void export_BarFile_ShouldInclude_NecessaryZipEntries() {
+        ZipEntry meta = barZipEntryMap.get("00_meta/");
+        assertNotNull(meta);
+        assertTrue(meta.isDirectory());
```

Comment:
In the future, it's better to compare expected xml content read from xml file placed in `test/resource/*` .

Reply:
This test is not unit test. It's combination or integration test over some public methods. But this time OK.